src/main/java/org/sg/campus/backing/CourseController.java

```diff
@@ -0,0 +1,169 @@
+package org.sg.campus.backing;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.faces.bean.ManagedBean;
+import javax.faces.bean.ManagedProperty;
+import javax.faces.bean.SessionScoped;
+
+import org.sg.campus.beans.ApplicationBean;
+import org.sg.campus.domain.Course;
+import org.sg.campus.domain.Topic;
+import org.sg.campus.util.JsfUtil;
+
+@ManagedBean
+@SessionScoped
+public class CourseController {
+
+	@ManagedProperty(value = "#{applicationBean}")
+	private ApplicationBean applicationBean;
+	private TopicController topicController;
+	private List<Course> courseList = new ArrayList<Course>();
+	private Course selectedCourse;
+
+	private List<Topic> allTopics = new ArrayList<Topic>();
+//	private List<Topic> selectedTopics;
+
+	private String newName;
+	private String newDescription;
+	private boolean newEnabled;
+	
+	public void updateCourseTopics() {
+		List<Topic> checkedTopics = new ArrayList<Topic>();
+		for (int i = 0; i < allTopics.size(); i++) {
+			final Topic topic = allTopics.get(i);
+			if (topic.isChecked()) {
+				checkedTopics.add(topic);
+			}
+		}
+		selectedCourse.setTopicList(checkedTopics);
+		System.out.println(selectedCourse);
+		cleanCheckboxes();
+	}
+	
+	public void cleanCheckboxes() {
+		for (int i = 0; i < allTopics.size(); i++) {
+			final Topic topic = allTopics.get(i);
+			topic.setChecked(false);
+		}
+	}
+	
+	public String cleanTopic(Course course) {
+		selectedCourse = course;
+		for (int i = 0; i < allTopics.size(); i++) {
+			final Topic topic = allTopics.get(i);
+			topic.setChecked(false);
+		}
+		System.out.println(selectedCourse);
```

Comment:
I would put a referenze in the printed string like: allTopics: (print topics)

Reply:
done